core/src/main/java/io/camunda/migrator/RuntimeMigrator.java

```diff
@@ -387,53 +394,63 @@ protected void activateMigratorJobs() {
+          LOGGER.debug("Collecting active descendant activity instances for activityId [{}]",
+              activityInstanceTree.getActivityId());
+          Map<String, FlowNode> activityInstanceMap = getActiveActivityIdsById(activityInstanceTree, new HashMap<>());
+          LOGGER.debug("Found {} active activity instances to activate", activityInstanceMap.size());
 
-        activityInstanceMap.forEach((activityInstanceId, flowNode) -> {
-          String activityId = flowNode.activityId();
-          var variableQuery = runtimeService.createVariableInstanceQuery().activityInstanceIdIn(activityInstanceId);
+          activityInstanceMap.forEach((activityInstanceId, flowNode) -> {
+            String activityId = flowNode.activityId();
+            var variableQuery = runtimeService.createVariableInstanceQuery().activityInstanceIdIn(activityInstanceId);
 
-          Map<String, Object> localVariables = new Pagination<VariableInstance>().batchSize(getBatchSize())
-              .query(variableQuery)
-              .context(context)
-              .toVariableMapSingleActivity();
+            Map<String, Object> localVariables = new Pagination<VariableInstance>().batchSize(getBatchSize())
+                .query(variableQuery)
+                .context(context)
+                .toVariableMapSingleActivity();
 
-          String subProcessInstanceId = flowNode.subProcessInstanceId();
-          if (subProcessInstanceId != null) {
-            localVariables.put("legacyId", subProcessInstanceId);
-          }
-
-          modifyProcessInstance.activateElement(activityId).withVariables(localVariables, activityId);
-        });
+            String subProcessInstanceId = flowNode.subProcessInstanceId();
+            if (subProcessInstanceId != null) {
+              localVariables.put("legacyId", subProcessInstanceId);
+            }
 
-        String batchUpdatedActivitiesErrorMessage = "Error while activating jobs";
-        callApi(() -> ((ModifyProcessInstanceCommandStep1.ModifyProcessInstanceCommandStep3) modifyProcessInstance).execute(), batchUpdatedActivitiesErrorMessage);
-        // no need to complete the job since the modification canceled the migrator job in the start event
+            modifyProcessInstance.activateElement(activityId).withVariables(localVariables, activityId);
+          });
+
+          String batchUpdatedActivitiesErrorMessage = "Error while activating jobs";
+          callApi(
+              () -> ((ModifyProcessInstanceCommandStep1.ModifyProcessInstanceCommandStep3) modifyProcessInstance).execute(),
+              batchUpdatedActivitiesErrorMessage);
+          // no need to complete the job since the modification canceled the migrator job in the start event
+        } else {
+          LOGGER.info("Process instance with key [{}] was externally started, skipping migrator job activation.", job.getProcessInstanceKey());
```

Comment:
❓ Is C8 process instance key the same as what we would call process instance ID in C7?

Reply:
I don't get this question. Are you proposing using the C8 ID instead of the C8 key?

Keys in C8 are automatically generated like IDs are in C7.

Reply:
> Are you proposing using the C8 ID instead of the C8 key?
No this already ansered my question :) I was more wondering about terminology, since we already learned the definition key / ID equivalency in C7/C8, just hadnt realised the same is true for instance terminology